File: kotlin/matrix/src/main/kotlin/Matrix.kt
class Matrix(private val matrixAsString: String) {
    private val matrix = matrixAsString
        .split("\n")
        .map {
            it.split(" ").map { str ->
                str.toInt()
            }
        }

    fun column(colNr: Int): List<Int> = matrix.map { it[colNr - 1] }

    fun row(rowNr: Int): List<Int> = matrix[rowNr - 1]
}

File: typescript/matrix/matrix.ts
export class Matrix {
  constructor(private input: string) {}

  get rows() {
    return this.input.split("\n").map((row) => row.split(" ").map(Number));
  }

  get columns() {
    const columLength = this.rows[0];
    return columLength.map((_, colIndex) =>
      this.rows.map((row) => row[colIndex])
    );
  }
}
